pkg/client/fake/client.go

```diff
@@ -527,21 +529,21 @@ func (c *fakeClient) List(ctx context.Context, obj client.ObjectList, opts ...cl
 		return err
 	}
 
-	ta, err := meta.TypeAccessor(o)
-	if err != nil {
-		return err
+	if _, isUnstructured := obj.(runtime.Unstructured); isUnstructured {
```

Comment:
Do we need the same for patch in l.874++?

(+ corresponding test cases)

Reply:
Great catch, updated it and added testcases for patch, update and delete